### ISSUE TITLE
Postgresql: Large double precision float values cause panic

### ISSUE THREAD

**Issue by DekzVt** [prisma/prisma#3923]
## Bug description

Prisma client panics on queries of float values over ~ 7.922e+28

## How to reproduce

I had originally tried this on version prisma version 2.1.3:
anything over ~9.223e+18 

1. Connect prisma to a table with a column of type "Double Precision" (float8)
2. Use a FindMany, FindOne, or $queryRaw to query on that table
3. Manually (sql) set a float value on one of the tables as 7.922e+28. This value should return from query as intended
4. Manually (sql) set a float value as 7.923e+28. This value will cause a panic with the error message:

```
"Invalid `prisma.stat.findOne()` invocation in\n/app/src/allTypes/queries/....
 PANIC: f64 is not a Decimal: Error { message: "Invalid decimal: overflow from scale mismatch" }
 This is a non-recoverable error which probably happens when the Prisma Query Engine has a panic."
```

## Expected behavior
Large decimal values should return as in step 3 of 'How to reproduce'.

## Prisma information

Table/Column in question looks like:
```
model Stat {
  ...
  healthPoints Float? @map("health_points")
  ...
  @map("stat")
}
```

## Environment & setup

- OS: Linux Debian (node:14-buster-slim docker image)
- Database: PostgreSQL
- Node.js version: 14
- Prisma version: 2.9.0

**Comment by DekzVt** [comment]
Apologies for delay @janpio 

> Also, not sure if I should make a separate issue for this, [...]

Definitely. Thanks!

_Originally posted by @janpio in https://github.com/prisma/prisma/issue_comments/706385716_

**Comment by pantharshit00** [comment]
Thanks @DekzVt for detailed report. I can confirm this.

@pimeys will look into this I guess when he will have some time.

**Comment by pimeys** [comment]
Hey. This is an issue of the decimal crate we use to represent floats, doubles, numbers and decimals. We hit the scale limits of that crate and to fix this, should switch to something like `bigdecimal` that doesn't have that kind of limitations.

This is of course a bigger project, and requires us to either switch to the SQLx backend that has support for the `bigdecimal`, or implementing the support to the `tokio-postgres` (and getting it merged).

So, in summary: we'll evaluating our best way forward with this. Thanks for reporting, we'll update this issue when we know more.
